storage/localstore/mode_set.go

```diff
@@ -393,3 +399,25 @@ func (db *DB) setUnpin(batch *leveldb.Batch, addr chunk.Address) (err error) {
 
 	return nil
 }
+
+// setReUpload adds a pinned chunk to the push index so that it can be re-uploaded to the network
+func (db *DB) setReUpload(batch *leveldb.Batch, addr chunk.Address) (err error) {
+	item := addressToItem(addr)
+
+	// only pinned chunks should be re-uploaded
+	// this also prevents a race condition: a non-pinned chunk could be gc'd after it's put in the push index
+	// but before it is actually re-uploaded to the network
+	_, err = db.pinIndex.Get(item)
+	if err != nil {
+		return err
+	}
+
+	retrievalDataIndexItem, err := db.retrievalDataIndex.Get(item)
+	if err != nil {
+		return err
+	}
+
+	item.StoreTimestamp = retrievalDataIndexItem.StoreTimestamp
+	// TODO: do we need to do anything about bin ids?
```

Comment:
I do not think that bin ids are relevant in this case.

Reply:
thanks, comment removed